src/app/server/Mdns.cpp

```diff
@@ -255,66 +381,82 @@ CHIP_ERROR Advertise(bool commissionableNode, CommissioningMode mode)
     return mdnsAdvertiser.Advertise(advertiseParameters);
 }
 
-/// Set MDNS commissioner advertisement
-CHIP_ERROR AdvertiseCommissioner()
+CHIP_ERROR MdnsServer::AdvertiseCommissioner()
 {
-    return Advertise(false /* commissionableNode */, CommissioningMode::kDisabled);
+    return Advertise(false /* commissionableNode */, chip::Mdns::CommissioningMode::kDisabled);
 }
 
-/// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode(CommissioningMode mode)
+CHIP_ERROR MdnsServer::AdvertiseCommissionableNode(chip::Mdns::CommissioningMode mode)
 {
     return Advertise(true /* commissionableNode */, mode);
 }
 
-/// (Re-)starts the minmdns server
-/// - if device has not yet been commissioned, then commissioning mode will show as enabled (CM=1, AC=0)
-/// - if device has been commissioned, then commissioning mode will reflect the state of mode argument
-void StartServer(CommissioningMode mode)
+void MdnsServer::StartServer(chip::Mdns::CommissioningMode mode)
 {
-    CHIP_ERROR err = chip::Mdns::ServiceAdvertiser::Instance().Start(&chip::DeviceLayer::InetLayer, chip::Mdns::kMdnsPort);
+    ChipLogDetail(Discovery, "Mdns StartServer mode=%d", static_cast<int>(mode));
 
-    err = app::Mdns::AdvertiseOperational();
+    ClearTimeouts();
+
+    CHIP_ERROR err;
+    err = chip::Mdns::ServiceAdvertiser::Instance().StopPublishDevice();
```

Comment:
This is the change that broke commissioning.

This removes all of the records, then restarts the server. Restarting the server is the only time unsolicited advertisements are sent. It is generally always the case that the device gets onto the network after the initial query and thus it is the unsolicited advertisements that provided the answer (via AdvertiseRecords()).